### PR TITLE
Feature newest post

### PR DESCRIPTION
**Pull request by Melissa-Martinez-XL**
## Ticket

* [Create outstanding post component](https://www.notion.so/xmartlabs/Create-outstanding-post-component-f9c5a958dfcf4eeaa94c7864527915f0?pvs=4)

## Type of change

* [ ] Fix
* [X] Story
* [ ] Chore

## Description of the change

I created the component which show the newest post for each category. Also, I added the excerpt.

## Screenshot/Execution

https://github.com/xmartlabs/xl-blog/assets/107500715/2e88a09e-4dfd-4732-844f-9c4cb2372d95

https://github.com/xmartlabs/xl-blog/assets/107500715/d9ecb458-e575-4823-91bc-746185ece5e5

Captura desde 2023-06-15 09-37-53

## Related PRs
Links to the related PRs

### CODE REVIEW

File: src/components/newest-post/newest-post.jsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  const urlImages = () => {
+    if (typeof window !== 'undefined') {
+      const imageUrl = window.location.origin;
+      return `${imageUrl}/${data.frontmatter.thumbnail}`;
+    }
+    return data.frontmatter.thumbnail;
+  }
+
+  return (
+    <>
```
Comment (lauta2912) [inline src/components/newest-post/newest-post.jsx:51]: not needed

File: src/layouts/layout.jsx [hunk 2]
```diff
@@ -97,11 +97,13 @@ function Layout({ children }) {
...
             className={classnames(styles.filterElement, "text__filter__grayFive", filter.name === category && styles.selectedLink)} 
             to={filter.name === 'all' ? '/' : `/categories/${filter.name}/`}>
-            {filter.displayName}
+              {filter.displayName}
           </Link>
         )}
     </div>
   );
+
+  category === undefined && setCategory("all");
```
Comment (lauta2912) [inline src/layouts/layout.jsx:106]: State updates can't be done in the render function. What case the category is `undefined` in?

Comment (Melissa-Martinez-XL) [reply src/layouts/layout.jsx:106]: I don't have a concrete answer, but, when I open a blog, and then click on the Xmartlabs logo (goes to the home page) the category is not selected. I did a console.log of category, and, in that scenario, I think, is the only case that category is undefined.

Comment (Melissa-Martinez-XL) [reply src/layouts/layout.jsx:106]: Also, if I have many renderers, Gtasby throws an error, in this case, it does not. But I don't know if my "solution" will have any effect on that.

File: src/components/newest-post/newest-post.jsx (hunk 1 above)
Comment (Melissa-Martinez-XL) [reply src/components/newest-post/newest-post.jsx:51]: Done

File: src/layouts/layout.jsx (hunk 2 above)
Comment (lauta2912) [reply src/layouts/layout.jsx:106]: Maybe you can check that the case in which the category is not defined is the one in line 76. if it is that, you can add `|| 'all'` at the end of the statement

**Review by lauta2912 — APPROVED** [review]

**Pull request merged**